### PR TITLE
[16.0][MIG] sale_order_product_assortment: Migration to version 16.0

### PR DESCRIPTION
**Pull request by carolinafernandez-tecnativa**
- Standard migration to v16
- Add new field allowed_product_tmpl_ids as in v16 on sales orders shows product_template_id 
- Change Setup to SetupClass tests
- Add test for new field allowed_product_tmpl_ids
- Supersedes https://github.com/OCA/sale-workflow/pull/2356 due to inactivity/changes requested not attended

@Tecnativa

TT44355

@pedrobaeza @carlosdauden

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please respect the commit history of the other PR, and just add on top your changes.

**Comment by carolinafernandez-tecnativa** [comment]
> Please respect the commit history of the other PR, and just add on top your changes.

Could you please check it now? Thanks :)

File: sale_order_product_assortment/__manifest__.py [hunk 1]
```diff
@@ -4,14 +4,18 @@
...
     "website": "https://github.com/OCA/sale-workflow",
     "author": "Tecnativa, Odoo Community Association (OCA)",
     "maintainers": ["CarlosRoca13"],
     "license": "AGPL-3",
     "installable": True,
-    "depends": ["sale", "product_assortment", "base_view_inheritance_extension"],
+    "depends": [
+        "sale",
+        "product_assortment",
+        "web_domain_field",
```
Comment (pedrobaeza) [inline sale_order_product_assortment/__manifest__.py]: Don't use this one, but create a binary field and use it as the source for the domain.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
The old migration commit is still not attributed to the previous contributor.

**Comment by yvaucher** [comment]
After the following commit:
https://github.com/OCA/product-attribute/commit/ac010741b1c8214c24bbcf7504242a263a36fe78

The Sale order view becomes really slow.

Plus, if we don't set an assortment to a customer it doesn't allow to select a product anymore on the sale order line.

**Comment by carolinafernandez-tecnativa** [comment]
ping @pedrobaeza

**Comment by pedrobaeza** [comment]
@carolinafernandez-tecnativa please check the comments from @yvaucher

**Comment by pedrobaeza** [comment]
@carolinafernandez-tecnativa please redo the migration from current v15, as this one is not valid due to the changes that makes it very slow.

**Comment by carolinafernandez-tecnativa** [comment]
> @carolinafernandez-tecnativa please redo the migration from current v15, as this one is not valid due to the changes that makes it very slow.

WIP, We need to merge this to redo.
https://github.com/OCA/product-attribute/pull/1682

**Comment by pedrobaeza** [comment]
Please redo the migration from scratch from current 15.0, as I see old or non optimized stuff (for example, there's no `allowed_product_tmpl_ids` in that branch) and a mix with the migration attempt that only confuses more the diff review.

**Pull request closed without merging**